### PR TITLE
Add `igraph` package

### PR DESCRIPTION
**Pull request by ntamas**
### Description

This PR adds support for the `igraph` package in Pyodide, as discussed in #3989

Currently igraph has a dependency on the `texttable` package, which is pure Python so it could be used as-is with Pyodide, but it is not in the `packages` folder yet. The `texttable` package is needed only for a very minor subset of igraph's functionality (pretty-printing vertex and edge attribute tables).

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add / update tests
- [x] Add new / update outdated documentation

### CODE REVIEW

**Comment by ntamas** [comment]
Please let me know how to proceed with the inclusion of the `texttable` package (if needed); see the PR description for more details.

**Comment by ntamas** [comment]
Seems like the package builds correctly now; one of the CI failures is due to a timeout and the other one is not related to the newly added package.

**Comment by ryanking13** [comment]
Thanks, @ntamas! Please update the changelog. 

> Please let me know how to proceed with the inclusion of the texttable package (if needed); see the PR description for more details.

If `texttable` is an optional dependency for `igraph`, then it is up to you whether or not to add it.

You can run `pyodide skeleton pypi texttable` to add a recipe for texttable, then you need to add the following in the `igraph` recipe.

```
requirements:
  run:
    - texttable
```

**Comment by ntamas** [comment]
Converting to draft because a new release of `igraph` (0.10.6) is about to happen in a few hours and it would be great to get this PR merged with that version instead.

**Comment by ntamas** [comment]
`igraph` updated to 0.10.6 and it builds fine locally on my machine, so let's see how this goes in CI.

File: packages/igraph/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package:
+  name: igraph
+  version: 0.10.5
```
Comment (ryanking13) [inline packages/igraph/meta.yaml:3]:
```suggestion
  version: 0.10.6
```

**Review by ryanking13 — APPROVED** [review]
Thanks a lot, @ntamas!

**Pull request merged**